addons/sms/tests/test_sms_template_access_rights.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import SavepointCase, new_test_user, users
+from odoo.tests import tagged
+from odoo.exceptions import AccessError
+
+
+@tagged('post_install')
+class TestSmsTemplateAccessRights(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user_admin = new_test_user(cls.env, login='user_admin', groups='base.group_system')
+        cls.basic_user = new_test_user(cls.env, login='basic_user', groups='base.group_user')
+        sms_enabled_models = cls.env['ir.model'].search([('is_mail_thread_sms', '=', True), ('transient', '=', False)])
+        vals = []
+        for model in sms_enabled_models:
+            vals.append({
+                'name': 'SMS Template ' + model.name,
+                'body': 'Body Test',
+                'model_id': model.id,
+            })
+        cls.sms_templates = cls.env['sms.template'].create(vals)
+
+    @users('basic_user')
+    def test_access_rights_user_sms_template(self):
+        # Check if a member of group_user can only read on sms.template
+        for sms_template in self.sms_templates.with_user(self.env.user):
```

Comment:
Instead of that `with_user(self.env.user)` that is very confusing, why not trying to read the templates?

```py
user_sms_templates = self.env['sms.template'].search([('id', 'in', self.sms_templates.ids)])
self.assertEqual(user_sms_templates.ids, self.sms_templates.ids)
for sms_template in user_sms_templates:
    # ...
```

As a bonus, you check that the basic_user can read those.
What do you think?